src/serial.rs

```diff
@@ -1862,6 +1862,16 @@ halUsart! {
     UART10: (uart10, apb2enr, 7, uart10en, pclk2),
 }
 
+impl<USART, PINS> fmt::Write for Serial<USART, PINS>
+where
+    Serial<USART, PINS>: serial::Write<u8>,
+{
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        let _ = s.as_bytes().iter().map(|c| block!(self.write(*c))).last();
```

Comment:
Hmm, I would really prefer a `for loop` in here, I have the impression that it's easier to read and more idiomatic in this case, and not sure if the fact that `last()` keeps track of the last valid value (i.e. a not `None` one) incurs a minimal performance hit, although that bother me less than the "easiness to read".

Reply:
```suggestion
        s.as_bytes().iter().for_each(|c| block!(self.write(*c)));
```

But I also prefer a for loop.

Reply:
Well, it didn't typecheck, sorry, block doesn't return ()

Reply:
No worries. How do you like the current version?